sr_demos/scripts/sr_demos/demo.py

```diff
@@ -46,29 +47,27 @@ def __init__(self, hand_commander, demo_joint_states, prefix):
                           "Talk to your Shadow representative to purchase some " +
                           "or use the keyboard to access this demo.")
 
-        # Zero values in dictionary for tactile sensors (initialized at 0)
-        self.force_zero = {"FF": 0, "MF": 0, "RF": 0, "LF": 0, "TH": 0}
         # Initialize values for current tactile values
         self.tactile_values = {"FF": 0, "MF": 0, "RF": 0, "LF": 0, "TH": 0}
-
+        # Zero values in dictionary for tactile sensors (initialized at 0)
+        self.reference_tactile_values = dict(self.tactile_values)
         self.zero_tactile_sensors()
 
     def zero_tactile_sensors(self):
         if self.get_tactiles() is not None:
-            rospy.sleep(0.5)
             rospy.logwarn('\nPLEASE ENSURE THAT THE TACTILE SENSORS ARE NOT PRESSED')
             input('\nPress ENTER to continue...')
-            rospy.sleep(1.0)
 
-            for x in range(1, 1000):
-                # Read current state of tactile sensors to zero them
+            # Collect 50 next samples and average them to filter out possible noise
+            accumulator = []
+            for i in range(0, 50):
```

Comment:
This is something we are likely (proof is that it has been changed here) to change in the future, so please make that a constant and avoid having magic numbers